src/Emulator/Extensions/Mocks/HPSHostController.cs

```diff
@@ -171,6 +172,37 @@ private byte[] GetBytesFromSlave(int count, TimeInterval timeInterval)
             return result;
         }
 
+        // This method is for getting OA1EN, 
+        // returns true when OA1EN is enabled.
+        private bool OA1ENEnabled(){
+            return currentDoubleSlave.ReadDoubleWord(0x08) >> 15 != 0x00;
```

Comment:
Due to a specific nature of the `HPSHostController` class, I'm wondering if it wouldn't be better for it to accept an instance of `STM32F7_I2C` as a constructor parameter instead of using the `IConnectable` interface.

With that explicit dependency no casting to `IDoubleWordPeripheral` is needed and it would be easier to access internal state of the I2C controller directly (i.e., reading properties, calling methods, etc.).

Reply:
I did think about that option, if it is okay to use `STM32F7_I2C` directly, this would save a lot of trouble indeed.

Reply:
Might be kinda trivial question but how do we pass in the `STM32F7_I2C` instance to the constructor? Or to be more specific, what method call can we retrieve the instance from the emulation object? 

Also, when I created an onject `private STM32F7_I2C currentSlave;` a compilation error will occur:
```
The type or namespace name `STM32F7_I2C' could not be found. Are you missing an assembly reference?
```
I have the dependency `using Antmicro.Renode.Peripherals.I2C;` inside the file but is there any other dependency that I'm missing here?

Reply:
> how do we pass in the STM32F7_I2C instance to the constructor?

My suggestion is to drop the `IConnectable` interface completely and use the following code in the `HPSHostController.cs` file:
```
[...]
  public static class HPSHostControllerExtensions
    {
        public static void AddHPSHostController(this Emulation emulation, STM32F7_I2C device, string name = "HPSHostController")
        {
            emulation.HostMachine.AddHostMachineElement(new HPSHostController(device), name);
        }
    }

    public class HPSHostController : IHostMachineElement
    {
        public HPSHostController(STM32F7_I2C device)
        {
              currentSlave = device;
        }
[...]
```
With this in place it should be enough to execute the following command in Monitor/resc to initiate the controller:
```
(machine-0) emulation AddHPSHostController sysbus.i2c1
```
(note that there is no `connector Connect` anymore).

> I have the dependency using Antmicro.Renode.Peripherals.I2C; inside the file but is there any other dependency that I'm missing here?

This was related to the fact that `HPSHostController` was a part of a different project and couldn't see the `STM32F7_I2C` type. It has been resolved in https://github.com/renode/renode/commit/79d959853bf7e2621e98a1ecce554d19700c6094. Sorry for the misleading comment, I was not aware that this change was technically problematic (but it's doable now).

Reply:
Thanks for the detailed explanation and new change! That worked!